app/directives/pdf/pdf.directive.js

```diff
@@ -11,48 +11,60 @@
         return {
             scope: {
                 uri: '=',
                 mode: '=',
-                hide: '=',
-                animate: '='
+                hide: '='
             },
             templateUrl: 'directives/pdf/pdf.Directive.html',
             restrict: 'AE',
             replace: 'true',
-            link: function(scope, element, attrs) {
+            link: function (scope, element/*, attrs*/) {
 
                 var HEIGHT = 'height';
                 var WIDTH = 'width';
                 var RESIZE = 'resize';
+                var SRC = 'src';
+                var params = "?secondary=false&amp;mixed=false#view=FitH&amp;toolbar=0&amp;statusbar=0&amp;messages=0&amp;navpanes=0";
                 var timer;
                 var pending = false;
                 var hidden = false;
-                var anim = angular.isDefined(attrs.animate);
+                var anim = false;
+
+                if (scope.uri) {
+                    element.attr(SRC, scope.uri + params);
+                }
+
+                element.css(HEIGHT, element.parent().height());
+                element.css(WIDTH, element.parent().width());
 
                 function hide() {
                     if (!hidden) {
                         hidden = true;
-                        if (anim) {
-                            element.removeClass('db-visible-pdf');
-                            element.addClass('db-hidden-pdf');
-                        }
+                        element.removeClass('db-visible-pdf');
+                        element.removeClass('db-visible-pdf-anim');
+                        element.addClass('db-hidden-pdf');
                     }
                 }
 
                 function show() {
                     pending = true;
                     if (!angular.isDefined(timer)) {
-                        timer = $interval(function() {
+                        timer = $interval(function () {
                             if (!pending) {
                                 $interval.cancel(timer);
                                 timer = undefined;
                                 element.css(HEIGHT, element.parent().height());
                                 element.css(WIDTH, element.parent().width());
+                                element.removeClass('db-hidden-pdf');
                                 if (anim) {
-                                    element.removeClass('db-hidden-pdf');
+                                    element.removeClass('db-visible-pdf');
+                                    element.addClass('db-visible-pdf-anim');
+                                }
+                                else {
+                                    element.removeClass('db-visible-pdf-anim');
                                     element.addClass('db-visible-pdf');
                                 }
                                 hidden = false;
```

Comment:
This in effect ignores the scope.hide given by directive's view.
If a view uses the directive and wants to control it's visibility, would it use ngIf/ngHide? If so, what purpose would  scope.hide attribute serve?
Should't the directive remain hidden always if scope.hide is true.